tests/unit/s2n_config_test.c

```diff
@@ -731,5 +732,120 @@ int main(int argc, char **argv)
+        {
+            DEFER_CLEANUP(struct s2n_config *config = s2n_minimal_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(config);
+            EXPECT_NULL(config->trust_store.trust_store);
+            EXPECT_FALSE(config->loaded_system_certs);
+
+            /* System certs can be loaded onto the minimal config */
+            EXPECT_SUCCESS(s2n_config_load_system_certs(config));
+            EXPECT_NOT_NULL(config->trust_store.trust_store);
+            EXPECT_TRUE(config->loaded_system_certs);
+
+            /* Attempting to load system certs multiple times on the same config should error */
+            for (int i = 0; i < 20; i++) {
+                EXPECT_FAILURE_WITH_ERRNO(s2n_config_load_system_certs(config), S2N_ERR_X509_TRUST_STORE);
+                EXPECT_TRUE(config->loaded_system_certs);
+            }
+        }
+
+        /* s2n_config_new should load system certs */
+        {
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(config);
+            EXPECT_NOT_NULL(config->trust_store.trust_store);
+            EXPECT_TRUE(config->loaded_system_certs);
+
+            /* Attempting to load system certs multiple times on the same config should error */
+            for (int i = 0; i < 20; i++) {
+                EXPECT_FAILURE_WITH_ERRNO(s2n_config_load_system_certs(config), S2N_ERR_X509_TRUST_STORE);
+                EXPECT_TRUE(config->loaded_system_certs);
+            }
+        }
+
+        /* The default config should load system certs */
+        {
+            struct s2n_config *config = s2n_fetch_default_config();
+            EXPECT_NOT_NULL(config);
+            EXPECT_TRUE(config->loaded_system_certs);
+        }
+
+        /* System certs can be loaded again after wiping the trust store */
+        {
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(config);
+
+            for (int i = 0; i < 20; i++) {
+                /* System certs were already loaded, so an attempt to load them should fail */
+                EXPECT_NOT_NULL(config->trust_store.trust_store);
+                EXPECT_TRUE(config->loaded_system_certs);
+                EXPECT_FAILURE_WITH_ERRNO(s2n_config_load_system_certs(config), S2N_ERR_X509_TRUST_STORE);
+
+                EXPECT_SUCCESS(s2n_config_wipe_trust_store(config));
+
+                /* The trust store is cleared after a wipe, so it should be possible to load system certs again */
+                EXPECT_FALSE(config->loaded_system_certs);
+                EXPECT_SUCCESS(s2n_config_load_system_certs(config));
+                EXPECT_TRUE(config->loaded_system_certs);
+            }
+        }
+
+        /* Ensure that system certs are properly loaded into the X509_STORE.
+         *
+         * The API to retrieve the contents of an X509_STORE wasn't added until OpenSSL 1.1.0.
```

Comment:
Might be helpful for future devs to specify the name of the API that's not available